### DISCUSSION TITLE
Transform:Sample 结果不符合预期

### DISCUSSION THREAD

**Discussion by xns1997** [antvis/G2#5676] [General]
## 描述
* 当遇到大规模数据可视化时，使用了G2内置lttb，性能未及预期，用法如下：
``` typescript
let scatter: MarkNode | null = null;
scatter = chart.point();
scatter.transform([{ type: 'sample', strategy: 'lttb', thresholds: 5000 }]);
scatter.tooltip(false);
scatter.animate(false);
```
## 问题
* 图像本身在观感上的确是被优化过的
* 内存占用并没有显著减少，大量不会用于绘制图像形状对象依然被生成
![截屏2023-10-24 14 54 27](https://github.com/antvis/G2/assets/16635242/88c2affe-2ad0-4411-a95b-ccb822eb878e)
<img width="400" alt="截屏2023-10-24 14 54 31" src="https://github.com/antvis/G2/assets/16635242/29e565ec-04b1-4a8c-a975-301d18cc28d6">

## 预期
* 减少不必要的内存占用
* 提高大数据场景下的渲染速度

**Comment by hustcc** [comment]
数据点有多少？方便搞一个在线 demo 不，我来调试一下！

**Reply by xns1997** [reply]
* 数据60000+条
* <a href="https://codesandbox.io/s/stoic-chandrasekhar-4tvcc2?file=/src/index.js">链接</a>
* 不排除是更改了其他配置导致的，但是没法在线上完整复现，因为用到的生成配置的逻辑很复杂。

**Reply by hustcc** [reply]
我修改这个 demo 中的配置，看起来是生效的啊，只不过对于散点图来说，使用 lttb 策略不合适，这个策略更适合折线图，看趋势的。

**Reply by xns1997** [reply]
在调试工具内会出现很多没有被显示到shape
